crates/trigger/src/loader.rs

```diff
@@ -56,6 +56,14 @@ impl Loader for TriggerLoader {
             )
         })?;
         let component = spin_componentize::componentize_if_necessary(&bytes)?;
+
+        let file_name = path
+            .file_name()
```

Comment:
As you note, there could be clashes here, and I believe with an OCI app this will be some unhelpful digest string.  Is there any downside to adding a component ID parameter to the `load_component` function, and using that to generate the dump file name?  I had a poke at it and the only internal caller seems to have ready access to the ID so the blast radius is low; I guess it depends on if there are any external callers.